WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m

```diff
@@ -121,185 +139,95 @@ - (void)viewDidLoad
-{
-    NSString *headingTitle = nil;
-    if (section == TableViewSectionTitle) {
-        headingTitle = self.blog.blogName;
-    } else if (section == TableViewSectionMobileSettings) {
-        headingTitle = NSLocalizedString(@"Settings", @"");
-    }  else if (section == TableViewSectionGeneralSettings) {
-        headingTitle = NSLocalizedString(@"General", @"");
-    }
-    return headingTitle;
-}
-
-- (UITableViewCell *)tableView:(UITableView *)tableView cellForSectionTitleInRow:(NSInteger)row
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForAccountSettingsInRow:(NSInteger)row
 {
     switch (row) {
-        case EditSiteRowURL: {
-            WPTextFieldTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:TextFieldCellIdentifier];
-            self.urlTextField = cell.textField;
-            cell.textLabel.text = NSLocalizedString(@"URL", @"");            
-            self.urlTextField.placeholder = NSLocalizedString(@"http://my-site-address (URL)", @"(placeholder) Help the user enter a URL into the field");
-            [self.urlTextField addTarget:self action:@selector(showSaveButton) forControlEvents:UIControlEventEditingChanged];
-            [self configureTextField:self.urlTextField asPassword:NO];
-            self.urlTextField.keyboardType = UIKeyboardTypeURL;
-            if (self.blog.url != nil) {
-                self.urlTextField.text = self.blog.url;
-                
-                // Make a margin exception for URLs since they're so long
-                cell.minimumLabelWidth = EditSiteURLMinimumLabelWidth;
-            } else {
-                self.urlTextField.text = @"";
+        case SiteSettingsAccountUsername: {
+            if (self.usernameTextCell) {
+                return self.usernameTextCell;
             }
-            
-            self.urlTextField.enabled = [self canEditUsernameAndURL];
-            [WPStyleGuide configureTableViewTextCell:cell];
-            
-            return cell;
-        } break;
-        case EditSiteRowUsername: {
-            WPTextFieldTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:TextFieldCellIdentifier];
-            
-            cell.textLabel.text = NSLocalizedString(@"Username", @"Label for entering username in the username field");
-            self.usernameTextField = cell.textField;
-            self.usernameTextField.placeholder = NSLocalizedString(@"Enter username", @"(placeholder) Help enter WordPress username");
-            [self.usernameTextField addTarget:self action:@selector(showSaveButton) forControlEvents:UIControlEventEditingChanged];
-            [self configureTextField:self.usernameTextField asPassword:NO];
-            if (self.blog.usernameForSite != nil) {
-                self.usernameTextField.text = self.blog.usernameForSite;
+            self.usernameTextCell = [[SettingTableViewCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:SettingCellIdentifier];
```

Comment:
If you're already passing the style when initializing here, there's no need to have a cell subclass, right?